lib/ansible/modules/remote_management/manageiq/manageiq_provider.py

```diff
@@ -0,0 +1,528 @@
+        required: true
+      metrics[userid]:
+        required: false
+        default: null
+        description:
+          - Provider's api endpoint authentication userid. defaults to None.
+      metrics[password]:
+        required: false
+        default: null
+        description:
+          - Provider's api endpoint authentication password. defaults to None.
+      metrics[auth_key]:
+        required: false
+        default: null
+        description:
+          - Provider's api endpoint authentication bearer token. defaults to None.
+      metrics[verify_ssl]:
+        required: false
+        default: true
+        description:
+          - Whether SSL certificates should be verified for HTTPS requests (deprecated). defaults to True.
+      metrics[security_protocol]:
+        required: false
+        default: None
+        choices: ['ssl-with-validation','ssl-with-validation-custom-ca','ssl-without-validation']
+        description:
+          - How SSL certificates should be used for HTTPS requests. defaults to None.
+      metrics[certificate_authority]:
+        required: false
+        default: null
+        description:
+          - The CA bundle string with custom certificates. defaults to None.
+'''
+
+EXAMPLES = '''
+  - name: Create a new provider in ManageIQ
+    manageiq_provider:
+      name: 'EngLab'
+      type: 'Openshift'
+      endpoints:
+        default:
+          auth_key: 'topSecret'
+          hostname: 'example.com'
+          port: 8443
+          verify_ssl: False
+        metrics:
+          role: 'prometheus'
+          hostname: 'example.com'
+          port: 443
+          verify_ssl: False
+      manageiq_connection:
+        url: 'http://127.0.0.1:3000'
+        username: 'admin'
+        password: 'smartvm'
+        verify_ssl: False
+
+  - name: Update an existing provider named 'EngLab'
+    manageiq_provider:
+      name: 'EngLab'
+      type: 'Openshift'
+      endpoints:
+        default:
+          auth_key: 'verySecret'
+          hostname: 'next.example.com'
+          port: 8443
+          verify_ssl: False
+        metrics:
+          hostname: 'next.example.com'
+          port: 443
+          verify_ssl: False
+      manageiq_connection:
+        url: 'http://127.0.0.1:3000'
+        username: 'admin'
+        password: 'smartvm'
+        verify_ssl: False
+
+  - name: Delete a provider in ManageIQ
+    manageiq_provider:
+      state: 'absent'
+      name: 'EngLab'
+      manageiq_connection:
+        url: 'http://127.0.0.1:3000'
+        username: 'admin'
+        password: 'smartvm'
+        verify_ssl: False
+'''
+
+RETURN = '''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.manageiq import ManageIQ, manageiq_argument_spec
+
+
+def supported_providers():
+    return dict(
+        Openshift=dict(
+            class_name='ManageIQ::Providers::Openshift::ContainerManager',
+            authtype='bearer',
+            metrics_role='hawkular',
```

Comment:
@cben @simon3z WDYT should be the default metrics_role 'hawkular' or 'prometheus' ?

we currently default to 'hawkular' but thinking of the future we should default to 'prometheus' ?